crates/gen/src/propertykey.rs

```diff
@@ -0,0 +1,25 @@
+use super::*;
+
+#[derive(Clone, PartialEq, Default)]
+pub struct PropertyKey {
+    pub fmtid: Guid,
+    pub pid: u32,
+}
+
+impl PropertyKey {
+    pub fn from_attributes<I: IntoIterator<Item = tables::Attribute>>(
+        attributes: I,
+    ) -> Option<Self> {
+        for attribute in attributes {
+            if attribute.name() == "PropertyKeyAttribute" {
+                let args = attribute.args();
+                return Some(Self {
+                    fmtid: Guid::from_args(&args),
+                    pid: args[11].1.unwrap_u32(),
+                });
+            }
+        }
+
+        None
```

Comment:
I think the [find_map](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.find_map) algorithm basically does this.

Reply:
I know you probably just copied what I did in `Guid::from_attributes` - I'm still learning. 😊